console/src/components/Services/Data/DataActions.js

```diff
@@ -574,10 +579,58 @@ const makeMigrationCall = (
     }
     customOnSuccess(data, globals.consoleMode, currMigrationMode);
   };
+  const retryMigration = (err = {}, errMsg = '') => {
+    dispatch(
+      showNotification({
+        title: errMsg,
+        level: 'error',
+        message: (
+          <p>
+            {getErrorMessage(err, '')}
+            <br />
+            <br />
+            Do you want to drop the dependent items as well?
+          </p>
+        ),
+        autoDismiss: 0,
+        action: {
+          label: 'Continue',
+          callback: () =>
+            makeMigrationCall(
```

Comment:
We can start thinking about using object for `makeMigrationCall` args. We could then have:
```js
const retryMigration = (/* */, args) => { ... }   // <- move outside

const makeMigrationCall = (args) => {
 const { ... } = args;
 
 // don't worry about creating retryMigration

  const onError = () => {
   ...
  retryMigration(allNeededArgs, args)
  }
  ...
}
```
Not necessary to do in this PR. Just a thought.

Reply:
Yeah I thought the same, makeMigration has a lot of arguments. will do this along with #4790

Reply:
simplified this with seperate function